aea/helpers/search/models.py

```diff
@@ -230,6 +238,9 @@ def _check_validity(self):
                 if len(self.value) > 0:
                     _type = type(next(iter(self.value)))
                     assert all(isinstance(obj, _type) for obj in self.value)
+            elif self.type == ConstraintTypes.DISTANCE:
```

Comment:
What motivates this? Can you elaborate?

Reply:
Since distance inherits from the ConstraintType : `class Distance(ConstraintType):` we need to provide the ConstraintTypes and the value to the super class : 
```
class ConstraintType:
    def __init__(self, type: Union[ConstraintTypes, str], value: Any):
```
Based on this the Distance class ends up like this : 
```
    def __init__(self, center: Location, distance: float) -> None:
        """
        Instantiate the ``Distance`` constraint.
        :param center: the center from where compute the distance.
        :param distance: the maximum distance from the center, in km.
        """
        self.center = center
        self.distance = distance
        super().__init__(ConstraintTypes.DISTANCE, [center, distance])
```